### PR TITLE
Companion PR for https://github.com/pytorch/pytorch/pull/83628

### PR DESCRIPTION
**Pull request by ezyang**
Signed-off-by: Edward Z. Yang <ezyang@fb.com>

### CODE REVIEW

File: torch_xla/csrc/aten_xla_type.cpp [hunk 1]
```diff
@@ -3199,7 +3195,8 @@ at::Tensor XLANativeFunctions::new_empty_strided(
 }
 
 at::Tensor XLANativeFunctions::narrow_copy(const at::Tensor& self, int64_t dim,
-                                           int64_t start, int64_t length) {
+                                           c10::SymInt start,
```
Comment (Krovatkin) [inline torch_xla/csrc/aten_xla_type.cpp:3198]: OoO, why did we also change `start` to `SymInt`. Do we have to switch all int arguments to `SymInt` for symint kernels?

**Review by Krovatkin — COMMENTED** [review]
@JackCaoG this is the change I mentioned last dynamic shapes sync, Ed was going to make. It's pretty boiler-platy.

File: torch_xla/csrc/aten_xla_type.cpp (hunk 1 above)
Comment (ezyang) [reply torch_xla/csrc/aten_xla_type.cpp:3198]: No. But start should be a SymInt as it is a "size-y" int. Notice that dim doesn't get changed.

**Review by JackCaoG — APPROVED** [review]
lgtm, thanks!

**Pull request merged**